### PR TITLE
Make some pubternal types in data protection

### PR DESCRIPTION
**Pull request by davidfowl**
- Make CngAuthenticatedEncryptorBase internal
- Left IActivator public because they are used in constructors the XmlKeyManager ctor

Contributes to #11312

### CODE REVIEW

**Comment by davidfowl** [comment]
Can I get an approval here.

File: src/DataProtection/DataProtection/test/Cng/CngAuthenticatedEncryptorBaseTests.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ public void Decrypt_HandlesEmptyCiphertextPointerFixup()
             Assert.Equal(new byte[] { 0x20, 0x21, 0x22 }, retVal);
         }
 
-        public abstract class MockableEncryptor : CngAuthenticatedEncryptorBase
+        internal abstract class MockableEncryptor : CngAuthenticatedEncryptorBase
```
Comment (anurse) [inline src/DataProtection/DataProtection/test/Cng/CngAuthenticatedEncryptorBaseTests.cs]: This is test code right? Still. I'm here for it :)

**Review by analogrelay — APPROVED** [review]
Couple of quick GitHub searches yield only forks of our own repo so it seems good to me.

**Comment by davidfowl** [comment]
Funny, I did the same search. All those damn forks 😄

**Pull request merged**